lib/smb2/file.rb

```diff
@@ -58,23 +79,23 @@ def inspect
   # @param length [Fixnum,nil] number of bytes to read. If this is nil, read
   #   the whole file, i.e. {#size} bytes.
   # @return [String] full contents of the remote file
-  def read(offset: 0, length: nil)
+  def read(length = nil, offset: self.pos)
```

Comment:
I assume `outbuf` is some String reuse optimization thing.  I'm fine with not supporting that if we can determine the actual need for it since some of these lower-level Ruby APIs are more C-ish than Ruby-ish.

The only compromise I could see is doing `*args` and supporting overloading, so it would support `read()`, `read(length)` or `def read(offset: self.pos, length: nil)`.  You could use YARD's `@overload` to document each of these calling conventions, but it makes unpacking the argument more complicated internally.